src/ops/stash.rs

```diff
@@ -0,0 +1,154 @@
+use super::{Action, OpTrait};
+use crate::{items::TargetData, prompt::PromptData, state::State, term::Term, Res};
+use derive_more::Display;
+use std::{process::Command, rc::Rc};
+use tui_prompts::State as _;
+
+#[derive(Default, Clone, Copy, PartialEq, Eq, Debug, Display)]
+#[display(fmt = "Stash")]
+pub(crate) struct Stash;
+impl OpTrait for Stash {
+    fn get_action(&self, _target: Option<&TargetData>) -> Option<Action> {
+        Some(stash_push_action("--include-untracked", ""))
+    }
+}
+
+#[derive(Default, Clone, Copy, PartialEq, Eq, Debug, Display)]
+#[display(fmt = "Stash index")]
+pub(crate) struct StashIndex;
+impl OpTrait for StashIndex {
+    fn get_action(&self, _target: Option<&TargetData>) -> Option<Action> {
+        Some(stash_push_action("--staged", ""))
+    }
+}
+
+#[derive(Default, Clone, Copy, PartialEq, Eq, Debug, Display)]
+#[display(fmt = "Stash working tree")]
+pub(crate) struct StashWorktree;
+impl OpTrait for StashWorktree {
+    fn get_action(&self, _target: Option<&TargetData>) -> Option<Action> {
+        // TODO: how to execute several commands?
+        // Ideally, for unstaged stashing we need to commit index, stash unstaged and then reset
+        // last commit. In the current implementation we include index in the stash despite index
+        // remains untouched.
```

Comment:
I tried this in magit, here's what it seems to be doing:

stash worktree:
```
  0 git … update-ref --create-reflog -m hello refs/stash 85a1205d04c2c49ebaf6b905de844e96eddfe55d 1ef91d084e4a5e8e3d29ada319e52e50ea801706
  0 git … checkout -- .
  ```
  
stash index:
```
  0 git … update-ref --create-reflog -m hello refs/stash abe8de6c219ffa784d7fe038e8ee95e3b20a9
  0 git … apply --reverse --cached --ignore-space-change -
  0 git … apply --reverse --ignore-space-change -
```

It's still running multiple commands though.
It is of course possible to do, but it'd be nice if it jacks in to the existing interfaces and displays some output.

Reply:
Perhaps we should comment this out and leave the TODO? And remove the references to 'StashWorktree' for now too.

Reply:
I not even sure why they run such complicated command, probably I don't understand something. I added running 3 commands, so the only issue right now is that we show only last run command, so I left TODO for that. It is working as expected so I think we can include that